src/mgmt/srv_drpc.c

```diff
@@ -355,6 +355,55 @@ create_pool_props(daos_prop_t **out_prop, char *owner, char *owner_grp,
 	return rc;
 }
 
+static int pool_create_fill_resp(Mgmt__PoolCreateResp *resp, uuid_t uuid, d_rank_list_t *svc_ranks)
+{
+	int			rc = 0;
+	d_rank_list_t	       *enabled_ranks = NULL;
+	daos_pool_info_t	pool_info = { .pi_bits = DPI_ENGINES_ENABLED | DPI_SPACE };
+
+	D_ASSERT(svc_ranks != NULL);
+	D_ASSERT(svc_ranks->rl_nr > 0);
+
+	rc = rank_list_to_uint32_array(svc_ranks, &resp->svc_reps, &resp->n_svc_reps);
+	if (rc != 0) {
+		D_ERROR("Failed to convert svc rank list: rc=%d\n", rc);
+		D_GOTO(out, rc);
+	}
+
+	D_DEBUG(DB_MGMT, "%d service replicas\n", svc_ranks->rl_nr);
+
+	rc = ds_mgmt_pool_query(uuid, svc_ranks, &enabled_ranks, &pool_info);
+	if (rc != 0) {
+		D_ERROR("Failed to query created pool: rc=%d\n", rc);
+		D_GOTO(out, rc);
+	}
```

Comment:
Is there a potential race here between the pool service having been created and being ready to handle queries? Just concerned about pool create operations appearing to fail because this query fails, leading to a pool cleanup and confusing behavior for the user.

Reply:
I agree, that there is a potential race condition as for example for the conversion of the ranklist convertion or the message packing. FMHO,, I do not thing that it will be confusing for the end user as it will just see that the pool creation as failed. Moreover, such situation should be very uncommon and should probably indicate that the pool created is unusable by the end user. To mitigate this race condition, I could also perform some retries  with random latencies between trials. By this way, we could handle transient networking issue.

Reply:
Well, I was under the impression that the implementation of `ds_mgmt_pool_query()` might handle transient communication issues with retries, but I'm not sure. I think it's worth confirming either way though. Mostly what I want to avoid is a situation where we do the pool query here immediately after creating it, and having a failure that causes the pool to be automatically cleaned up. 

@kccain, @liw: Any advice here?

Reply:
A pool query to a pool service that is not ready yet should (I think) result in a cart RPC reply with status -DER_NOTLEADER and a hint of what rank to try to contact next in the "search for the leader" logic. ds_mgmt_pool_query() (though running on an engine) does have client RPC logic to rechoose another pool service replica rank and retry the RPC when that -DER_NOTLEADER response is seen. And this should continue until a successful reply, or some other error is seen (one that I assume would be unrelated to any pool start up latency).

Some questions come to mind though. In cases where scm size is specifically given by the admin in the pool create request was it ever promised (from an API/interface perspective) that the exact number of bytes allocated would be returned in the pool create response? In other cases where a single size is given, and scm and nvme sizes are calculated based on a ratio I guess it may make more sense to return the calculated (or maybe even actual allocated) amount to be informative to the admin about what was allocated. In the scm_size given in the request, I see in the code why a subsequent list-pools/query may report a different amount, given the single target size calculation that rounds up to a multiple of 4KB.

Reply:
Although using a pool query may not be the most economical way (for we could gather the info via the responses of the RPCs that create the pool, I think it should work.

Reply:
From what I remember when I did this stuff, the call to the RPC `ds_mgmt_create_pool()` and its subsequent function call does not return the really allocated size. Thus, I used the pool query to avoid too much intrusive code modification. However, from what I understand, such modifications are worthwhile and thus I will update my PR to not have to do some extra RPC call.

- [ ] Remove RPC poll_query call

Reply:
I missed a ")" in my last comment. Added. The current pool-query-based approach is fine with me.

Reply:
After more investigation with @mjmac, we agreed that the pool-query-base was not the best approach (in terms of performance) but was acceptable to fix this issue: it should not introduce reliability issue.